src/v/config/configuration.cc

```diff
@@ -4504,6 +4504,12 @@ configuration::configuration()
       "cluster for data replication.",
       meta{.needs_restart = needs_restart::no, .visibility = visibility::user},
       false)
+  , rpc_request_timeout_ms(
+      *this,
+      "rpc_request_timeout_ms",
```

Comment:
note on naming: should we call it internal somewhere in the name?

"internal_rpc_request_timeout_ms" or some such (indicating it doesn't affect any user facing stuff).

just a suggestion, feel free to skip.